### PR TITLE
Driver controls

### PR DESCRIPTION
**Pull request by james-ward**
First pass at driver controls
* Add co-driver gamepad
* Turret disable
* Ball eject through intake

### CODE REVIEW

File: robot.py [hunk 1]
```diff
@@ -107,49 +110,73 @@ def teleopInit(self):
...
             if self.indexer.intaking:
                 self.indexer.disable_intaking()
+                self.indexer.raise_intake()
             else:
                 self.indexer.enable_intaking()
-
-    def handle_spinner_inputs(self, joystick: wpilib.Joystick) -> None:
+                self.indexer.lower_intake()
+
+        if gamepad.getAButtonPressed():
```
Comment (mlists) [inline robot.py]: I think you want getAButton() otherwise this will only run for 1 control loop

File: components/turret.py [hunk 2]
```diff
@@ -206,6 +209,10 @@ def is_ready(self) -> bool:
         )
 
     def execute(self) -> None:
+        if self.disabled:
+            self.motor.stop()
```
Comment (auscompgeek) [inline components/turret.py]: Isn't the method called `disable` or `stopMotor`?

File: controllers/shooter.py [hunk 3]
```diff
@@ -46,12 +46,26 @@ def __init__(self) -> None:
...
         self.distance = None
+        self.disabled_flash: int = 0
 
     def execute(self) -> None:
         super().execute()
         self.update_LED()
 
     def update_LED(self) -> None:
+        # Flash if turret and shooter are disabled
+        if self.shooter.disabled:
```
Comment (mlists) [inline controllers/shooter.py]: This comment seems to be wrong, the lights don't check if the turret is disabled

Comment (james-ward) [reply controllers/shooter.py]: They get disabled at the same time

File: components/turret.py (hunk 2 above)
Comment (james-ward) [reply components/turret.py]: Yes. Done in a bus to Canberra with no internet.

File: robot.py (hunk 1 above)
Comment (james-ward) [reply robot.py]: Correct

**Pull request merged**